zonemaster/zonemaster-ldns
# Ignore incomplete rr

## Purpose

This PR makes sure that we're not assuming all RRs come with all fields included. Out-of-the-box ldns returns such RRs even though they're incomplete. If we're unlucky we could be accessing such non-existing fields and that could lead to assertion errors or unexpected `undef` values.

## Context

I found this when investigating #110.

## Changes

* Introduces Zonemaster::LDNS::RR::check_rd_count() that validates the rdata field count for a record.
* Changes Zonemaster::LDNS::Packet::{answer,authority,additional}() to filter out incomplete RRs.
* The full set of RRs is still available through Zonemaster::LDNS::Packet::{answer,authority,additional}_unfiltered().

## How to test this PR

Just run `make test`. New unit test are added to verify that incomplete RRs are filtered out. The old tests verify that complete RRs are not filtered out.
